bundle/EventListener/ScheduledVisibilityListener.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\EventListener;
+
+use Ibexa\Contracts\Core\Repository\Events\Content\PublishVersionEvent;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class ScheduledVisibilityListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly ScheduledVisibilityService $scheduledVisibilityService
+    ) {}
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PublishVersionEvent::class => ['onPublishVersion'],
+        ];
+    }
+
+    public function onPublishVersion(PublishVersionEvent $event): void
+    {
+        $content = $event->getContent();
+        if ($this->scheduledVisibilityService->accept($content)) {
```

Comment:
It's good that `accept()` method is public, so it's possible to check whether the Content is acceptable without invoking `toggleVisibility()`, but it should not be necessary to call it in the consumer code; `toggleVisibiltity()` should call it itself.

Reply:
fixed in https://github.com/netgen/ibexa-scheduled-visibility/pull/1/commits/402cf61ff9afabab87e4abd134d9f7ffd5343cc9